MapleServer2/Managers/FieldManager.cs

```diff
@@ -711,26 +793,608 @@ public FieldObject(int objectId, T value)
+            {
+                Status status = new(skillCast, ObjectId, ObjectId, 1);
+                //StatusHandler.Handle(Value.Session, status);
+            }
+        }
+
+        public virtual void RecoverHp(int amount)
+        {
+            if (amount <= 0)
+            {
+                return;
+            }
+
+            lock (Stats)
+            {
+                Stat stat = Stats[StatId.Hp];
+                if (stat.Total < stat.Bonus)
+                {
+                    stat.Increase(Math.Min(amount, stat.Bonus - stat.Total));
+                }
+            }
+        }
+
+        public virtual void ConsumeHp(int amount)
+        {
+            if (amount <= 0)
+            {
+                return;
+            }
+
+            lock (Stats)
+            {
+                Stat stat = Stats[StatId.Hp];
+                stat.Decrease(Math.Min(amount, stat.Total));
+            }
+        }
+
+        public virtual void RecoverSp(int amount)
+        {
+            if (amount <= 0)
+            {
+                return;
+            }
+
+            lock (Stats)
+            {
+                Stat stat = Stats[StatId.Spirit];
+                if (stat.Total < stat.Bonus)
+                {
+                    stat.Increase(Math.Min(amount, stat.Bonus - stat.Total));
+                }
+            }
+        }
+
+        public virtual void ConsumeSp(int amount)
+        {
+            if (amount <= 0)
+            {
+                return;
+            }
+
+            lock (Stats)
+            {
+                Stat stat = Stats[StatId.Spirit];
+                Stats[StatId.Spirit].Decrease(Math.Min(amount, stat.Total));
+            }
+        }
+
+        public virtual void RecoverStamina(int amount)
+        {
+            if (amount <= 0)
+            {
+                return;
+            }
+
+            lock (Stats)
+            {
+                Stat stat = Stats[StatId.Stamina];
+                if (stat.Total < stat.Bonus)
+                {
+                    Stats[StatId.Stamina].Increase(Math.Min(amount, stat.Bonus - stat.Total));
+                }
+            }
+        }
+
+        public virtual void ConsumeStamina(int amount)
+        {
+            if (amount <= 0)
+            {
+                return;
+            }
+
+            lock (Stats)
+            {
+                Stat stat = Stats[StatId.Stamina];
+                Stats[StatId.Stamina].Decrease(Math.Min(amount, stat.Total));
+            }
+        }
+
+        public virtual void Damage(DamageHandler damage)
+        {
+            Stat health = Stats[StatId.Hp];
+            health.Decrease((long) damage.Damage);
+            if (health.Total <= 0)
+            {
+                Perish();
+            }
+        }
+
+        public virtual void Perish()
+        {
+            IsDead = true;
+        }
+    }
+
+    private class Character : FieldActor<Player>
```

Comment:
Actually, can a folder of actors just be made, and every private class thrown into a separate class inside that folder?